sugarscape.py

```diff
@@ -893,50 +899,48 @@ def updateRuntimeStats(self):
-        self.agentsBorn = 0
-        self.runtimeStats["agentWealthCollected"] = agentWealthCollected
-        self.runtimeStats["agentWealthTotal"] = agentWealthTotal
-        self.runtimeStats["agentWealthBurnRate"] = agentWealthBurnRate
-        self.runtimeStats["agentMeanTimeToLive"] = agentMeanTimeToLive
-        self.runtimeStats["environmentWealthCreated"] = environmentWealthCreated
-        self.runtimeStats["environmentWealthTotal"] = environmentWealthTotal
-
-        self.runtimeStats["agentTotalMetabolism"] = agentTotalMetabolism
-        self.runtimeStats["totalSickAgents"] = totalSickAgents
+        runtimeStats = {"agentAgingDeaths": agentAgingDeaths, "agentCombatDeaths": agentCombatDeaths, "agentDiseaseDeaths": agentDiseaseDeaths, "agentMeanTimeToLive": agentMeanTimeToLive,
+                        "agentStarvationDeaths": agentStarvationDeaths, "agentTotalMetabolism": agentTotalMetabolism, "agentWealthBurnRate": agentWealthBurnRate,
+                        "agentWealthCollected": agentWealthCollected, "agentWealthTotal": agentWealthTotal, "maxWealth": maxWealth, "meanAge": meanAge, "meanAgeAtDeath": meanAgeAtDeath,
+                        "meanConflictHappiness": meanConflictHappiness, "meanFamilyHappiness": meanFamilyHappiness, "meanHappiness": meanHappiness, "meanHealthHappiness": meanHealthHappiness,
+                        "meanMetabolism": meanMetabolism, "meanMovement": meanMovement, "meanSocialHappiness": meanSocialHappiness, "meanTradePrice": meanTradePrice, "meanWealth": meanWealth,
+                        "meanWealthHappiness": meanWealthHappiness, "meanVision": meanVision, "minWealth": minWealth, "population": numAgents, "sickAgents": sickAgents, "tradeVolume": tradeVolume
+                        }
+
+        if group == None:
+            self.runtimeStats["agentsBorn"] = self.agentsBorn
+            self.runtimeStats["agentsReplaced"] = self.agentsReplaced
+            self.runtimeStats["environmentWealthCreated"] = environmentWealthCreated
+            self.runtimeStats["environmentWealthTotal"] = environmentWealthTotal
+            self.runtimeStats["giniCoefficient"] = self.updateGiniCoefficient()
+            self.runtimeStats["timestep"] = self.timestep
+            self.runtimeStats["totalMetabolismCost"] += totalMetabolismCost
+            self.runtimeStats["totalWealthLost"] += totalWealthLost
+            self.agentsBorn = 0
+            self.agentsReplaced = 0
+            self.deadAgents = []
+        else:
+            # Convert keys to Pythonic case scheme
+            groupString = group if notInGroup == False else "control"
+            groupStats = {}
+            for key in runtimeStats.keys():
+                groupKey = groupString + key[0].upper() + key[1:]
+                groupStats[groupKey] = runtimeStats[key]
+            runtimeStats = groupStats
+
+        for key in runtimeStats.keys():
+            self.runtimeStats[key] = runtimeStats[key]
+
+    def updateStats(self):
```

Comment:
Consider renaming `updateStats` and/or `updateRuntimeStats` to better differentiate between them if you can come up with something more descriptive — they're both runtime stats. Maybe `updateRuntimeStats` could include something group-related.

Reply:
Done.